pallets/asset-index/src/types.rs

```diff
@@ -33,16 +33,36 @@ pub enum AssetAvailability {
     Saft,
 }
 
+/// Metadata for an asset
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug)]
+pub struct AssetMetadata {
+    pub name: Vec<u8>,
+    pub symbol: Vec<u8>,
+    pub decimals: u8,
+}
```

Comment:
Use fixed `Vec<u8>` or associated types?

Reply:
We need something like this https://github.com/paritytech/substrate/blob/master/frame/assets/src/lib.rs#L264 that configures the metdata https://github.com/paritytech/substrate/blob/master/frame/assets/src/types.rs#L92

so we need a `StringLimit` type and make these vecs bounded